services/user/Webmail/service/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+use std::collections::HashMap;
+
+use psibase::check;
+use psibase::services::accounts::Wrapper as AccountsSvc;
+use psibase::services::r_events::Wrapper as REventsSvc;
+use psibase::AccountNumber;
+use psibase::HttpReply;
+use psibase::HttpRequest;
+
+fn validate_user(acct: AccountNumber) {
+    check(
+        AccountsSvc::call().exists(acct),
+        format!("account '{}' DNE", acct).as_str(),
+    );
+}
+
+fn make_query(req: &HttpRequest, sql: &str) -> HttpRequest {
+    return HttpRequest {
+        host: req.host.clone(),
+        rootHost: req.rootHost.clone(),
+        method: String::from("POST"),
+        target: String::from("/sql"),
+        contentType: String::from("application/sql"),
+        body: sql.to_string().into(),
+    };
+}
+
+fn parse_query(query: String) -> HashMap<String, String> {
+    let mut params: HashMap<String, String> = HashMap::new();
+
+    let itr = query.split("&");
+    for p in itr {
+        let idx = p.find("=").unwrap();
+        let kv = p.split_at(idx);
+        params.insert(kv.0.to_string(), kv.1.trim_start_matches('=').to_string());
+    }
+    params
+}
+
+fn serve_rest_api(request: &HttpRequest) -> Option<HttpReply> {
+    if request.method == "GET" {
+        if !request.target.starts_with("/messages") {
+            return None;
+        }
+
+        let query_start = request.target.find('?');
+        if query_start.is_none() {
+            return None;
+        }
+        let query_start = query_start.unwrap();
+
+        let query = request.target.split_at(query_start + 1).1;
+        let params = crate::parse_query(String::from(query));
+
+        let mut s_clause = String::new();
+        let s_opt = params.get(&String::from("sender"));
+        if let Some(s) = s_opt {
+            validate_user(AccountNumber::from(s.as_str()));
+            s_clause = format!("sender = '{}'", s);
+        }
+
+        let mut r_clause = String::new();
+        let r_opt = params.get(&String::from("receiver"));
+        if let Some(r) = r_opt {
+            validate_user(AccountNumber::from(r.as_str()));
+            r_clause = format!("receiver = '{}'", r);
+        }
+
+        if s_opt.is_none() && r_opt.is_none() {
+            return None;
+        }
+
+        let mut where_clause: String = String::from("WHERE ");
+        if s_opt.is_some() {
+            where_clause += s_clause.as_str();
+        }
+        if s_opt.is_some() && r_opt.is_some() {
+            where_clause += " AND ";
+        }
+        if r_opt.is_some() {
+            where_clause += r_clause.as_str();
+        }
+
+        let mq = make_query(
+            request,
+            format!(
+                "SELECT * FROM \"history.webmail.sent\" {} ORDER BY ROWID",
+                where_clause
+            )
+            .as_str(),
```

Comment:
`make_query` can just take a `String` and avoid going from `String` -> `&str` -> `String`